Tracebacks from collection errors are not pruned

This is a regression in pytest 7.4 from commit fcada1ea4763c0e3471cd58ac4b89e7c874e6264.

Given a test module

```py
raise Exception('foo')
```

Expected output:

```
_______ ERROR collecting yy/test_y.py _______
yy/test_y.py:1: in <module>
    raise Exception('foo')
E   Exception: foo
```

Actual output:

```
_______ ERROR collecting yy/test_y.py _______
src/_pytest/runner.py:340: in from_call
    result: Optional[TResult] = func()
src/_pytest/runner.py:371: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
src/_pytest/python.py:579: in collect
    self._inject_setup_module_fixture()
src/_pytest/python.py:593: in _inject_setup_module_fixture
    self.obj, ("setUpModule", "setup_module")
src/_pytest/python.py:307: in obj
    self._obj = obj = self._getobj()
src/_pytest/python.py:576: in _getobj
    return importtestmodule(self.path, self.config)
src/_pytest/python.py:528: in importtestmodule
    mod = import_path(path, mode=importmode, root=config.rootpath)
src/_pytest/pathlib.py:567: in import_path
    importlib.import_module(module_name)
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1204: in _gcd_import
    ???
<frozen importlib._bootstrap>:1176: in _find_and_load
    ???
<frozen importlib._bootstrap>:1147: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
src/_pytest/assertion/rewrite.py:175: in exec_module
    exec(co, module.__dict__)
yy/test_y.py:1: in <module>
    raise Exception('foo')
E   Exception: foo
```

--- a/changelog/11710.bugfix.rst
+++ b/changelog/11710.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed tracebacks from collection errors not getting pruned.

--- a/src/_pytest/nodes.py
+++ b/src/_pytest/nodes.py
@@ -579,7 +579,7 @@ class Collector(Node, abc.ABC):
             ntraceback = traceback.cut(path=self.path)
             if ntraceback == traceback:
                 ntraceback = ntraceback.cut(excludepath=tracebackcutdir)
-            return excinfo.traceback.filter(excinfo)
+            return ntraceback.filter(excinfo)
         return excinfo.traceback
 
 

--- a/testing/test_collection.py
+++ b/testing/test_collection.py
@@ -16,7 +16,6 @@ from _pytest.nodes import Item
 from _pytest.pathlib import symlink_or_skip
 from _pytest.pytester import HookRecorder
 from _pytest.pytester import Pytester
-from _pytest._code.code import ExceptionChainRepr, ReprTraceback
 
 
 def ensure_file(file_path: Path) -> Path:

--- a/testing/test_collection.py
+++ b/testing/test_collection.py
@@ -16,6 +16,7 @@ from _pytest.nodes import Item
 from _pytest.pathlib import symlink_or_skip
 from _pytest.pytester import HookRecorder
 from _pytest.pytester import Pytester
+from _pytest._code.code import ExceptionChainRepr, ReprTraceback
 
 
 def ensure_file(file_path: Path) -> Path:
@@ -344,6 +345,29 @@ class TestPrunetraceback:
         )
         result = pytester.runpytest(p)
         result.stdout.fnmatch_lines(["*ERROR collecting*", "*header1*"])
+
+    def test_collection_error_traceback_is_clean(self, pytester: Pytester) -> None:
+        """When a collection error occurs, the report traceback doesn't contain
+        internal pytest stack entries.
+
+        Issue #11710.
+        """
+        pytester.makepyfile(
+            """
+            raise Exception("LOUSY")
+            """
+        )
+        result = pytester.runpytest()
+        result.stdout.fnmatch_lines(
+            [
+                "*ERROR collecting*",
+                "test_*.py:1: in <module>",
+                '    raise Exception("LOUSY")',
+                "E   Exception: LOUSY",
+                "*= short test summary info =*",
+            ],
+            consecutive=True,
+        )
 
 
 class TestCustomConftests:

